src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,28 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+
+    public final String value;
+    protected boolean isPrivate;
+
+    public Contact(String Contact,boolean isPrivate){
+        this.isPrivate = isPrivate;
+        String trimmedContact = Contact.trim();
+        this.value = trimmedContact;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
If the `equals()` method is overridden in subclass then `hashCode()` has to also be overridden.